packages/cli/generators/app/templates/test/acceptance/ping.controller.acceptance.ts.ejs

```diff
@@ -25,7 +26,47 @@ describe('PingController', () => {
   });
 
   it('invokes GET /ping', async () => {
-    await client.get('/ping?msg=world').expect(200);
+    const res = await client.get('/ping?msg=world').expect(200);
+    expect(res.body).to.containEql({greeting: 'Hello from LoopBack'});
+  });
+
+  it('exposes OpenAPI spec at /openapi.json', async () => {
```

Comment:
The test name feels misleading to me. We are not testing the entire OpenAPI spec, only the spec of `GET /ping` endpoint here. How about:

```ts
it('provides a valid OpenAPI spec for "GET /ping"', async () => {
 // ...
});
```

Personally, I don't think we have to make an HTTP request to test that, I would move this test to `src/test/integration` and obtain the spec via `app.restServer.getApiSpec()`. Such test will be much more resilient against possible changes of `/openapi.json` location in the future.

I see the value in a test obtaining the spec via HTTP. I think such test should go to `test/acceptance/application.acceptance.ts` and verify that the returned spec is valid.

```ts
import {validateApiSpec} from '@loopback/testlab';

describe('TodoApplication' /* use the real app class name */, () => {
  it('exposes valid OpenAPI spec at /openapi.json', async () => {
    const res = await client.get('/openapi.json').expect(200);
    await validateApiSpec(res.body);
  });
});
```

I am ok with leaving such test out of scope of this pull request if you prefer.